go/enclave/vkhandler/vk_handler.go

```diff
@@ -22,23 +22,23 @@ type VKHandler struct {
 	publicViewingKey *ecies.PublicKey
 }
 
-// New creates a new viewing key handler
-// checks if the signature is valid
-// as well if signature matches account address
-// todo (@ziga) - function now accepts both old and new messages
+// New creates a new viewing key handler checks if the signature is valid as well if signature matches account address
 func New(requestedAddr *gethcommon.Address, vkPubKeyBytes, accountSignatureHexBytes []byte) (*VKHandler, error) {
-	// Recalculate the message signed by MetaMask.
-	msgToSign := viewingkey.GenerateSignMessageOG(vkPubKeyBytes, requestedAddr)
-
-	// We recover the key based on the signed message and the signature.
-	recoveredAccountPublicKey, err := crypto.SigToPub(accounts.TextHash([]byte(msgToSign)), accountSignatureHexBytes)
+	userID := hex.EncodeToString(crypto.Keccak256Hash(vkPubKeyBytes).Bytes())
```

Comment:
can we add a comment on every step here.
It's not easy to follow if you don't have this fresh

Reply:
comments added